lib/asciidoctor/abstract_node.rb

```diff
@@ -221,6 +221,13 @@ def roles
     end
   end
 
+  # Public: A convenience method that adds the given role directly to this node
+  def add_role(name)
+    role_attribute = @attributes['role']
+    roles = role_attribute ? role_attribute.split(' ') : []
+    @attributes['role'] = (roles.dup << name).join(' ')
```

Comment:
To be clear, it's the `split` that creates the local detached reference in the current implementation. So there's no risk we are modifying the original reference.